### PR TITLE
Sakiii 3746

### PR DESCRIPTION
**Pull request by stuartf**
https://jira.sakaiproject.org/browse/SAKIII-3746

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -23,6 +23,11 @@ gem 'capistrano'
 # Allow for multi-stage deploys with Capistrano
 gem 'capistrano-ext'
 
+group :test do
```
Comment (croby) [inline Gemfile]: We probably want to include this in dev too

File: test/factories.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  f.stars 5
+  f.association :widget
+end
+
+Factory.define :screenshot do |f|
+  f.association :widget
+end
+
+Factory.define :state do |f|
+  f.title "accepted"
```
Comment (croby) [inline test/factories.rb:46]: Can we create all 3 states here, instead of just creating the one?

File: test/factories.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+end
+
+Factory.define :rating do |f|
+  f.review "This is a super awesome widget, I use it every day and get a warm fuzzy feeling"
+  f.stars 5
+  f.association :widget
+end
+
+Factory.define :screenshot do |f|
+  f.association :widget
```
Comment (croby) [inline test/factories.rb:41]: Is there a way to actually upload a screenshot here?

File: test/factories.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+Factory.define :category do |f|
+  f.sequence(:title) { |n| "Category #{n}" }
+end
+
+Factory.define :icon do |f|
+  f.association :widget
```
Comment (croby) [inline test/factories.rb:16]: Can we actually upload an icon here?

File: test/factories.rb (hunk 2 above)
Comment (stuartf) [inline test/factories.rb:46]: These are just the defaults you get when you do `Factory(:state)`, you can get one that's "pending" by doing `Factory(:state, :title => "pending")`

**Pull request merged**